### PR TITLE
[FIX]  ref #343 fix and factorisation of sorting method for notifications

### PR DESCRIPTION
**Pull request by MLE-Astek**

### CODE REVIEW

File: portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js [hunk 1]
```diff
@@ -77,18 +70,26 @@ var NotificationTable = React.createClass({
...
             state.n = n;
             state.currentSort = currentSort;
             component.setState(state);
-        };
+        }.bind(this);
+    },
+    sortElement: function(a,b){
+        var currentSort = this.state.currentSort;
+
+        if (typeof a[currentSort.prop] == typeof a[currentSort.prop]) {
```
Comment (bobeal) [inline portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js]: typo ?

File: portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js [hunk 2]
```diff
@@ -77,18 +70,26 @@ var NotificationTable = React.createClass({
...
+            if (typeof a[currentSort.prop] == "number") {
+                return (a[currentSort.prop] - b[currentSort.prop]) * currentSort.dir;
+            }
+            if (typeof a[currentSort.prop] == "string") {
+                return a[currentSort.prop].localeCompare(b[currentSort.prop]) * currentSort.dir;
```
Comment (bobeal) [inline portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js]: You didn't replace the use of localeCompare (see https://github.com/ozwillo/ozwillo-portal/issues/343#issuecomment-161368999) ?

File: portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js (hunk 1 above)
Comment (MLE-Astek) [inline portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js]: yes … wrong copy-paste

**Comment by bobeal** [comment]
Btw, when you sort by message in reverse order, notifications without a message appear first. Wouldn't it be more logical to have them appear last ?

File: portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js (hunk 2 above)
Comment (MLE-Astek) [inline portal-parent/ozwillo-portal-front/src/main/resources/public/js/notifications.jsx.js]: I haven't see the tbroyer comment. I'm searching for  a replacement

**Comment by bobeal** [comment]
Merged but still having notifications without an associated application appear first when ordering by application, whatever the sort order is (see screenshot). And the weird ordering described here : https://github.com/ozwillo/ozwillo-portal/issues/343#issuecomment-162539982.

screen shot 2015-12-13 at 12 47 35

**Pull request merged**